src/components/MainPage.tsx

```diff
@@ -1,14 +1,88 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { fetchFares } from '../helpers/ApiCallHelper';
 import TrainBoard from './TrainBoard';
 import UserInputFormArea from './UserInputFormArea';
 
+export type StationInfo = {
+    name: string;
+    latitude: number;
+    longitude: number;
+    aliases: string[];
+    id: number;
+    crs: string;
+    nlc: string;
+    isGrownStation : boolean; 
+    isSilverSeekStation : boolean;
+}
+
+export type CompactStationInfo = {
+    displayName: string;
+    nlc : string;
+    crs : string;
+}
+
+export type JourneyInfo = {
+    originStation: CompactStationInfo;
+    destinationStation: CompactStationInfo;
+    departureTime: Date;
+    arrivalTime: Date;
+    status: string;
+    journeyDurationInMinutes: number;
+    isFastestJourney: boolean;
+    isOvertaken: boolean;
+}
+
 const MainPage: React.FC = () => {
 
+    const [departureStation, setDepartureStation] = useState<StationInfo>();
+    const [destinationStation, setDestinationStation] = useState<StationInfo>();
+    const [availableJourneys, setAvailableJourneys] = useState<JourneyInfo[]>([]);
+
+    const handleBodyFromFareFetch = (body: any) => {
```

Comment:
What does handleBodyFromFareFetch do, could you be more specific than the word handle?